src/dune_rules/melange_rules.ml

```diff
@@ -179,6 +179,24 @@ let add_rules_for_libraries ~dir ~scope ~emit_stanza_dir ~sctx ~requires_link
                 (Path.Build.to_string emit_stanza_dir)
             ]
       in
+      let () =
+        let modes = Lib_info.modes info in
+        match modes.melange with
+        | false ->
+          User_error.raise ~loc:(fst mel.libraries)
+            [ Pp.textf
+                "The library %S was added as a dependency of the melange.emit \
+                 stanza with target %S, but this library is not compatible \
+                 with melange. To fix this, either:"
+                (Lib_name.to_string (Lib_info.name info))
+                mel.target
+            ; Pp.textf "- add (modes melange) to the library stanza"
+            ; Pp.textf
+                "- or remove the library from the libraries field in the \
+                 melange.emit stanza"
+            ]
```

Comment:
cc @christinerose I feel like there should be some full stops here but I am not sure. What do you think?

Reply:
It's difficult for me to see what's going on here. As sentences on their own, they would look like this: 

The library %S was added as a dependency of the melange.emit stanza with target %S, but this library is not compatible with melange. To fix this, either add (modes melange) to the library stanza or remove the library from the libraries field in the melange.emit stanza.

Still those are cumbersome, but I can't see how to tighten them without changing the meaning. If you'd like to try, we can collaborate on revising them.